File: python/module-2/lesson-4/run_eval.py
import asyncio
import sys
from pathlib import Path
from langsmith import aevaluate

# Add the agent directory to the path
agent_dir = Path(__file__).resolve().parent.parent.parent / "officeflow-agent"
sys.path.insert(0, str(agent_dir))

from agent_v4 import chat, load_knowledge_base
from eval_schema_check import sql_starts_with_schema_check

DATASET_NAME = "officeflow-dataset"
EXPERIMENT_PREFIX = "schema-check-eval"


async def target(inputs: dict) -> dict:
    return await chat(inputs["question"])


async def main():
    # Load knowledge base before running evals
    await load_knowledge_base(str(agent_dir / "knowledge_base"))

    results = await aevaluate(
        target,
        data=DATASET_NAME,
        evaluators=[sql_starts_with_schema_check],
        experiment_prefix=EXPERIMENT_PREFIX,
        max_concurrency=2,
    )
    print("Evaluation complete! View results at the link above.")


if __name__ == "__main__":
    asyncio.run(main())

File: ts/module-2/lesson-4/run_eval.ts
/**
 * Run the schema-before-query evaluator against the officeflow-dataset.
 *
 * Usage:
 *    npx tsx run_eval.ts
 */
import "dotenv/config";
import { evaluate } from "langsmith/evaluation";
import { uuid7 } from "langsmith";
import { chat, loadKnowledgeBase, setThreadId } from "../../officeflow-agent/agent_v5.js";
import { schemaBeforeQuery } from "./eval_schema_check.js";
import { fileURLToPath } from "url";
import * as path from "path";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function setup(): Promise<void> {
  /** Load knowledge base before running evals. */
  const agentDir = path.resolve(__dirname, "..", "..", "officeflow-agent");
  const kbDir = path.join(agentDir, "knowledge_base");
  await loadKnowledgeBase(kbDir);
}

function runAgent(inputs: Record<string, any>): Promise<Record<string, any>> {
  /** Invoke the agent with a fresh thread_id each time. */
  setThreadId(uuid7().toString());
  return chat(inputs.question);
}

await setup();

const results = await evaluate(runAgent, {
  data: "officeflow-dataset",
  evaluators: [schemaBeforeQuery],
  experimentPrefix: "schema-check-v5",
});
